app/models/message.rb

```diff
@@ -233,6 +234,20 @@ def ensure_processed_message_content
     self.processed_message_content = message_content&.truncate(150_000)
   end
 
+  # fetch the in_reply_to message and set the external id
+  def ensure_in_reply_to
```

Comment:
Can we make this a service and use it everywhere else? The attributes and the actions are common, right? 

1/ Get message by source_id
2/ If present, populate in_reply_to and in_reply_to_external_id

Reply:
You mean, moving this out as a concern?

Reply:
Saw the changes https://github.com/chatwoot/chatwoot/issues/6964, please let me know if you want to pair up on externalizing this?